kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/KafkaUtils.java

```diff
@@ -78,55 +76,40 @@ public static void produceMessagesWithCmd(String deployNamespace, List<String> e
     }
 
     /**
-     * Consume messages.
+     * Gets consumer records.
      *
-     * @param topicName the topic name
-     * @param name the name
-     * @param namespace the deploy namespace
-     * @param clientJob the client job
-     * @param message the message
-     * @param numOfMessages the num of messages
-     * @param timeout the timeout
-     * @return the string
+     * @param topicName the topic name (needed because kaf does not return topic name in json message)
+     * @param logRecords the log records
+     * @return the consumer records
      */
-    public static String consumeMessages(String topicName, String name, String namespace, Job clientJob, String message, int numOfMessages, Duration timeout) {
-        LOGGER.atInfo().setMessage("Consuming messages from '{}' topic").addArgument(topicName).log();
-        kubeClient().getClient().batch().v1().jobs().inNamespace(namespace).resource(clientJob).create();
-        String podName = KafkaUtils.getPodNameByLabel(namespace, "app", name, Duration.ofSeconds(30));
-        String log;
-        try {
-            log = await().alias("Consumer waiting to receive messages")
-                    .ignoreException(KubernetesClientException.class)
-                    .atMost(timeout)
-                    .until(() -> {
-                        if (kubeClient().getClient().pods().inNamespace(namespace).withName(podName).get() != null) {
-                            return kubeClient().logsInSpecificNamespace(namespace, podName);
-                        }
-                        return null;
-                    }, m -> StringUtils.countMatches(m, message) == (numOfMessages));
-        }
-        catch (ConditionTimeoutException e) {
-            log = kubeClient().logsInSpecificNamespace(namespace, podName);
-            LOGGER.atInfo().setMessage("Timeout! Received: {}").addArgument(log).log();
+    public static List<ConsumerRecord<String, String>> getConsumerRecords(String topicName, List<String> logRecords) {
+        List<ConsumerRecord<String, String>> records = new ArrayList<>();
+        for (String logRecord : logRecords) {
+            if (!TestUtils.isJSONValid(logRecord)) {
+                continue;
+            }
+            int partition = Integer.parseInt(TestUtils.getValueFromJson(logRecord, "partition"));
+            int offset = Integer.parseInt(TestUtils.getValueFromJson(logRecord, "offset"));
+            String key = TestUtils.getValueFromJson(logRecord, "key");
+            String payload = TestUtils.getValueFromJson(logRecord, "payload");
+            ConsumerRecord<String, String> consumerRecord = new ConsumerRecord<>(topicName, partition, offset, key, payload);
```

Comment:
Can't you use objectmapper to map straight to the ConsumerRecord?  Make the CR an interface and use a specialisation per client implementation.  Annotate the fields with jsonproperty to enable the deserialisation.

Reply:
We would have an issue with that since not all the json elements has the same name for all the kafka clients, i.e. `timestamp` is `ts` in kcat.
Is there a way to say that an element can be named in a different way for the deserialisation?

Reply:
Jackson has an alias annotation, that would probably function but it is not nice from an information hiding point of view.  You want to keep implementation concerns scoped as privately as possible. 

I was suggesting making a type hierarchy for ConsumerRecord <- KcatConsumerRecord etc to keep the precise field mapping is a private concern.

Reply:
Yeap, that would be good to have an specific ConsumerRecord for each one. I will do that. Thanks!